### PR TITLE
Add inmem storage

### PR DESCRIPTION
**Pull request by jefferai**
This satisfies Storage using an in-memory implementation, good for ephemeral use-cases.

### CODE REVIEW

File: storage.go [hunk 1]
```diff
@@ -42,3 +42,11 @@ type Storage interface {
 	// disambiguate what to list, and can be a nil pointer to the type.
 	List(context.Context, proto.Message) ([]string, error)
 }
+
+// CleanableStorage is an interface that can optionally be implemented by storage
+// implementations to indicate that there is a cleanup function that can be run
+// when storage usage is complete.
+type CleanableStorage interface {
+	// Cleanup is a cleanup function
+	Cleanup(context.Context) error
```
Comment (jimlambrt) [inline storage.go:51]: Looks like you're using this as `t.Cleanup(storage.Cleanup)`.  Unfortunately, t.Cleanup only takes `func()` without a parameter.

You could provide a `TestCleanup()` func which wraps the existing `Cleanup(context.Context)` for this specific use-case.

Comment (jimlambrt) [reply storage.go:51]: Or you could simply: `t.Cleanup(func() { ts.Cleanup(ctx) })`

File: storage/inmem/inmem.go [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+	ts.RLock()
+	defer ts.RUnlock()
+
+	select {
+	case <-ctx.Done():
+		return nil, ctx.Err()
+	default:
+	}
+
+	select {
```
Comment (jimlambrt) [inline storage/inmem/inmem.go:232]: why is this `select` repeated?

File: storage/inmem/inmem.go [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+func (ts *Storage) Load(ctx context.Context, msg nodeenrollment.MessageWithId) error {
+	const op = "nodeenrollment.storage.inmem.(Storage).Load"
+	if err := types.ValidateMessage(msg); err != nil {
+		return fmt.Errorf("(%s) given message cannot be loaded: %w", op, err)
+	}
+	subPath, err := subPathFromMsg(msg)
+	if err != nil {
+		return fmt.Errorf("(%s) given message cannot be loaded: %w", op, err)
+	}
+	err = ts.loadValue(ctx, msg.GetId(), subPath, msg)
```
Comment (jimlambrt) [inline storage/inmem/inmem.go:89]: perhaps add a test, where the msg is not found?

**Review by jimlambrt — CHANGES_REQUESTED** [review]
Looks great, just a few minor questions/suggestions.

File: storage.go (hunk 1 above)
Comment (jefferai) [reply storage.go:51]: That's exactly what my test does...

File: storage/inmem/inmem.go (hunk 2 above)
Comment (jefferai) [reply storage/inmem/inmem.go:232]: Oops, will fix.

File: storage/inmem/inmem.go (hunk 3 above)
Comment (jefferai) [reply storage/inmem/inmem.go:89]: Added

**Pull request merged**